org.openjdk.jmc.console.ext.agent/src/main/java/org/openjdk/jmc/console/ext/agent/manager/wizards/PresetEditingWizardConfigPage.java

```diff
@@ -65,13 +84,183 @@ public void createControl(Composite parent) {
 		Composite container = new Composite(sc, SWT.NONE);
 		sc.setContent(container);
 
-		// TODO: create manager control here
-		container.setLayout(new FillLayout());
-		new Label(container, SWT.NONE).setText("TODO: create config page control here");
+		GridLayout layout = new GridLayout();
+		container.setLayout(layout);
+
+		{
+			Composite fileNameContainer = createFileNameContainer(container);
+			GridData gd = new GridData(SWT.FILL, SWT.FILL, true, false);
+			fileNameContainer.setLayoutData(gd);
+		}
+
+		{
+			Control separator = createSeparator(container);
+			GridData gd = new GridData(SWT.FILL, SWT.FILL, true, false);
+			separator.setLayoutData(gd);
+		}
+
+		{
+			Composite globalConfigContainer = createGlobalConfigContainer(container);
+			GridData gd = new GridData(SWT.FILL, SWT.FILL, true, false);
+			globalConfigContainer.setLayoutData(gd);
+		}
+
+		populateUi();
+		bindListeners();
 
 		sc.setExpandHorizontal(true);
 		sc.setExpandVertical(true);
 		sc.setMinSize(container.computeSize(SWT.DEFAULT, SWT.DEFAULT));
 		setControl(sc);
 	}
+
+	private Composite createFileNameContainer(Composite parent) {
+		Composite container = new Composite(parent, SWT.NONE);
+		int cols = 5;
+		GridLayout layout = new GridLayout(cols, false);
+		layout.horizontalSpacing = 8;
+		container.setLayout(layout);
+
+		createFileNameInput(container, cols);
+
+		return container;
+	}
+
+	private void createFileNameInput(Composite parent, int cols) {
+		Label label = createLabel(parent, LABEL_FILE_NAME);
+		label.setLayoutData(new GridData(SWT.BEGINNING, SWT.CENTER, false, false));
+
+		fileNameText = createText(parent);
```

Comment:
Perhaps add a message to prompt the user with an example filename such as `fileNameText.setMessage("newPreset.xml"`);`

Reply:
I think `.setMessage()` is more like a description of what's except, not an example of it.

As for examples, as soon as model classes' default constructor set a default value to these properties, the value will be filled in by the `populateUi()` methods.

Reply:
Ok - in the event wizard right now default values are initially blank (or zero values), and so the message appears when a new event is being added.  When the field is populated by the user this new value will show up instead of the message.
This is my preference, because as a user you don't have to delete any text to enter in your preferred text.  However, both ways are correct so we should just choose a uniform way to go about this I guess.

Reply:
The reason I'd like to see a non-empty default for fields like `id` and `filename` is that it can be configured to avoid collisions, things automatically appending an incrementing id (eg. `new_preset (1).xml`, and `new_preset.event.2` when `new_preset.event.1` already exists).

However, doing so requires model instances to be created using factory methods instead of `new` keyword, that is, `preset.createEvent():IEvent` instead of `new Event()`. If you like this proposal I can comment with more details under #58

Reply:
Ok yes, this makes sense to me

Reply:
This was implemented in 2584177